Update pkg & add vite.config

happycollision opened the pull request:
> I am creating this as a starting point to learn how to upgrade from Snowpack to Vite. I figure if I can help you get this going, then I'll know better how to update all mine. I figure the first step in issue #4 is to just get it running on Vite.
>
> I made a few assumptions (especially regarding `not-svelte-kit`) that may be wrong for Windows users.
>
> More commits to come!

This is the result of running `npm init svelte@next` and then reverting
things that shouldn't change.

Next step is to actually adjust vite config and remove snowpack config.

happycollision committed:
Update pkg & add vite.config

This is the result of running `npm init svelte@next` and then reverting
things that shouldn't change.

Next step is to actually adjust vite config and remove snowpack config.

happycollision commented:
I am also not quite sure yet if the `vice.config.js` needs to actually be `.cjs`, but that is what it gave me.

happycollision committed:
-- WIP

This should eventually be squashed into the previous commit. For now,
these changes *seem* correct, though I still cannot run the app.

happycollision commented:
FYI, I had to rename snowpack config because sveltekit was detecting it and warning about a necessary switch to vite.

GrygrFlzr commented:
Oh, interesting, I hadn't started this because the latest SvelteKit was failing out-of-the-box if I enabled TypeScript.

> I made a few assumptions (especially regarding not-svelte-kit)

That part looks fine to me - `not-svelte-kit` is dependent on `@sveltejs/kit`'s version and the latest builds have fixed most of the stuff I monkey-patched anyway.

`adapt` probably should be removed since `svelte-kit build` now does it right after it runs the build process.

> I am also not quite sure yet if the `vite.config.js` needs to actually be `.cjs`, but that is what it gave me.

No, `.js` is correct - `.cjs` refers to the usage of CommonJS import and export syntax (e.g. `require` and `module.exports`, whereas `.js` implies it's using ESM imports and export syntax.

---

Additional notes:
- The last commit I did actually broke things for the Vercel deploy so I might need to replace it with Vite's glob imports.
- `snowpack.config.cjs` would eventually need removal to avoid `svelte-kit build` warning us about its existence
- Not entirely sure why but the Vercel CI is failing at... `@`?

```
09:32:13.098  	> svelte-kit build
09:32:13.244  	Unexpected option extensions
09:32:13.414  	vite v2.0.5 building for production...
09:32:13.481  	transforming...
09:32:14.109  	✓ 12 modules transformed.
09:32:14.109  	> Unexpected character '@' (2:0)
09:32:14.109  	[rollup-plugin-dynamic-import-variables] Unexpected character '@' (2:0)
09:32:14.109  	file: /vercel/workpath0/src/routes/_global.pcss:2:0
09:32:14.110  	SyntaxError: Unexpected character '@' (2:0)
09:32:14.110  	    at Object.pp$4.raise (/vercel/workpath0/node_modules/rollup/dist/shared/rollup.js:15857:13)
09:32:14.110  	    at Object.pp$9.getTokenFromCode (/vercel/workpath0/node_modules/rollup/dist/shared/rollup.js:17650:8)
09:32:14.110  	    at Object.getTokenFromCode (/vercel/workpath0/node_modules/rollup/dist/shared/rollup.js:19350:20)
09:32:14.111  	    at Object.pp$9.readToken (/vercel/workpath0/node_modules/rollup/dist/shared/rollup.js:17351:15)
09:32:14.111  	    at Object.pp$9.nextToken (/vercel/workpath0/node_modules/rollup/dist/shared/rollup.js:17342:15)
09:32:14.111  	    at Object.parse (/vercel/workpath0/node_modules/rollup/dist/shared/rollup.js:13407:8)
09:32:14.111  	    at Function.parse (/vercel/workpath0/node_modules/rollup/dist/shared/rollup.js:13429:35)
09:32:14.111  	    at Object.Graph.contextParse [as parse] (/vercel/workpath0/node_modules/rollup/dist/shared/rollup.js:19134:70)
09:32:14.111  	    at Object.transform (/vercel/workpath0/node_modules/vite/dist/node/chunks/dep-e0f09032.js:33778:27)
09:32:14.111  	    at /vercel/workpath0/node_modules/rollup/dist/shared/rollup.js:19049:25
```

GrygrFlzr commented:
Ah, now that I read over the logs again, it's probably dying at the lack of PostCSS parsing.

happycollision committed:
-- WIP correct config intellisense

happycollision commented:
Yeah, it's a start is all. Bouncing over to my project to see if I can learn anything there to bring back here.

happycollision committed:
-- WIP

Add search directory for postcss. I think this also turns it on.

GrygrFlzr commented on vite.config.js:
```diff
@@ -0,0 +1,18 @@
+    resolve: {
+        alias: {
+            $components: resolve('src/components'),
+            $app: resolve('.svelte/assets/runtime/app')
```
I believe this alias for `$app` is not necessary - SvelteKit already does it for us internally.

happycollision committed:
-- WIP drop unnecessary alias

happycollision committed:
-- WIP remove css config

I take it back. Testing with an intentional error in the config. It is loaded without any help. So not sure where the `@` error is coming from

happycollision commented:
🕺
